hawkular-command-gateway/hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusEndpointProcessors.java

```diff
@@ -222,8 +221,8 @@ protected void onBasicMessage(BasicMessageWithExtraData<BasicMessage> messageWit
     @Inject
     private BusCommandContextFactory commandContextFactory;
 
-    @Resource(name = Constants.CONNECTION_FACTORY_JNDI)
-    private ConnectionFactory connectionFactory;
+    @Inject
+    private BusConnectionFactoryProvider connectionFactoryProvider;
```

Comment:
> You could actually inject the ConnectionFactory directly

I am akin to think that it would bring the original problem of the ConnectionFactory not being ready under the JNDI name on time. I think so, because `BusEndpointProcessors` is `@ApplicationScoped` and `@Observes @Initialized(ApplicationScoped.class)`. Due to that, the attempt to resolve the JNDI name would happen (1) much earlier than in the present implementation and (2) the `BusConnectionFactoryProvider.getConnectionFactory()` may bolck for some time and I am not sure how much the CDI would accept such a potentially long blocking. WDYT?